esahg/esahg.php

```diff
@@ -1,149 +1,170 @@
-
-	return $file_hashes;
+<?php
+/*
+Plugin Name: Exploit Scanner Auto Hash Generator
+Plugin URI: https://github.com/mattyrob
+Description: Checks for and then automatically generates hash files for use in the Exploit Scanner plugin for WordPress.
+Version: 1.0b
+Author: Matthew Robinson
+Author URI: https://github.com/mattyrob
+Licence: GPL3
+Donate link: https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&amp;hosted_button_id=2387904
+Text Domain: esahg
+*/
+
+define( 'ESAHGDIR', trailingslashit( plugin_basename( dirname( __FILE__ ) ) ) );
+define( 'ESAHGPATH', WP_CONTENT_DIR . '/' . ESAHGDIR );
+add_action( 'admin_menu', 'esahg_admin' );
+
+function esahg_admin() {
+	load_plugin_textdomain( 'eashg', 'wp-content/plugins/' . ESAHGDIR, '/' . ESAHGDIR );
+	add_management_page( __( 'Exploit Scanner Auto Hash Generator', 'eashg' ), __( 'Exploit Scanner Auto Hash Generator', 'eashg' ), apply_filters( 'eashg_capability', 'manage_options' ), 'eashg', 'eashg_menu' );
+} // end eashg_admin()
+
+function eashg_menu() {
+	global $wpdb;
+	echo '<div class="wrap">';
+	echo '<h2>' . __( 'Exploit Scanner Automatic Hash File Generator', 'esahg' ) . '</h2>';
+	if ( ! is_plugin_active( 'exploit-scanner/exploit-scanner.php' ) ) {
+		$exit_msg = sprintf( __( '%s is not active on this site, please download and activate it.', 'esahg' ), '<a href="https://wordpress.org/plugins/exploit-scanner/">Exploit Scanner</a>' );
+		exit( $exit_msg );
+	}
+	require( ABSPATH . WPINC . '/version.php' );
+	$es_path = WP_CONTENT_DIR . '/plugins/exploit-scanner/';
+	$hash_filename = 'hashes-' . $wp_version . '.php';
+	$file = $es_path . $hash_filename;
+
+	// instantiate WordPress file system for later
+	$creds = request_filesystem_credentials( site_url() . '/wp-admin/', '', false, false, array() );
+	if ( ! WP_Filesystem( $creds ) ) {
+		echo '<p>' . __( 'Unable to enable the WordPress Filesystem.', 'esahg' ) . '</p>';
+		exit;
+	}
+	global $wp_filesystem;
+
+	if ( ! is_readable( $file ) ) {
+		// attempt to get hashes from Wordpress.org API
+		$response = wp_remote_get( 'https://api.wordpress.org/core/checksums/1.0/?version=' . $GLOBALS['wp_version'] );
+		$checksums = json_decode( wp_remote_retrieve_body( $response ) );
+
+		if ( false !== $checksums->{'checksums'}->{$GLOBALS['wp_version']} ) {
+			$hashes_arr = array();
+			foreach ( $checksums->{'checksums'}->{$GLOBALS['wp_version']} as $filename => $checksum ) {
+				if ( false === strstr( $filename, 'wp-content/plugins/' ) && false === strstr( $filename, 'wp-content/themes/' ) ) {
+					$hashes_arr[ $filename ] = $checksum;
+				}
+			}
+			$hashes = '<?php' . "\r\n" . '$filehashes = array(' . "\r\n";
+			foreach ( $hashes_arr as $filename => $hash ) {
+				$hashes .= "'" . $filename . "' => '" . $hash . "',\r\n";
+			}
+			$hashes .= ");\r\n?>";
+		} else {
+			// fall back to zip file download if checksums are not returned
+			global $download_to;
+			$zip_filename = 'wordpress-' . $wp_version . '.zip';

```

Comment:
I one sentence: it needs a rewrite. But I use wp-cli on servers and I don't stick to high quality shared hosting tools.
Have you meant that I would implement these?